activemodel/lib/active_model/type/helpers/numeric.rb

```diff
@@ -29,7 +29,7 @@ def non_numeric_string?(value)
             # 'wibble'.to_i will give zero, we want to make sure
             # that we aren't marking int zero to string zero as
             # changed.
-            value.to_s !~ /\A-?\d+\.?\d*\z/
+            value.to_s !~ /\A[-+]?\d+\.?\d*([eE][-+]\d+)?\z/
```

Comment:
Can you use `match?` to avoid creating unused match object and unused captures?

```diff
diff --git a/activemodel/lib/active_model/type/helpers/numeric.rb b/activemodel/lib/active_model/type/helpers/numeric.rb
index 16e14f9e5f..c9519dc908 100644
--- a/activemodel/lib/active_model/type/helpers/numeric.rb
+++ b/activemodel/lib/active_model/type/helpers/numeric.rb
@@ -29,7 +29,7 @@ def non_numeric_string?(value)
             # 'wibble'.to_i will give zero, we want to make sure
             # that we aren't marking int zero to string zero as
             # changed.
-            value.to_s !~ /\A-?\d+\.?\d*\z/
+            !/\A[-+]?\d+\.?\d*(?:[eE][-+]\d+)?\z/.match?(value.to_s)
           end
       end
     end
```

Reply:
Thank you @kamipo.
I found `5e2` is also valid expression, so I'll change regex to `/\A[-+]?\d+\.?\d*(?:[eE][-+]?\d+)?\z/`

I also found "E-notataion" is valid for `to_d` and `to_f` but not `to_i`:

```
irb(main):001:0> require 'bigdecimal'
=> true
irb(main):002:0> require 'bigdecimal/util'
=> true
irb(main):003:0> '5e+2'.to_d
=> 0.5e3
irb(main):004:0> '5e+2'.to_f
=> 500.0
irb(main):005:0> '5e+2'.to_i
=> 5
```

Should I overwrite this method for `AM::Type::Integer`?